### ISSUE TITLE
Car Display Shows Unsupported

### ISSUE THREAD

**Issue by camper8080** [oandrew/ipod-gadget#15]
Connected to USB port in a Honda Accord.

Car display shows Unsupported after a bit. Used swap config option a few different ways but same issue.

[tracefile.log](https://github.com/oandrew/ipod-gadget/files/4149294/tracefile.log)

Any help is appreciated.

Thank you!

**Comment by camper8080** [comment]
[tracefile2.log](https://github.com/oandrew/ipod-gadget/files/4150309/tracefile2.log)

**Comment by oandrew** [comment]
try this branch:
https://github.com/oandrew/ipod/pull/7

**Comment by camper8080** [comment]
Not a coder by trade. Any help on switching branches? I may be able to hack my way through it. Thanks again for your time and making this project.

**Comment by oandrew** [comment]
just merged it to master

**Comment by oandrew** [comment]
`git pull`

**Comment by camper8080** [comment]
Thank you. I'll give it a try as soon as I can.

**Comment by camper8080** [comment]
I ended up wiping the Pi and reinstalling everything. The speaker test works for a bit then stops, car goes back to unsupported. Not sure if that is normal or not. Perhaps it's working now and now I need to present some audio. Plan is to use Shairport-sync. Let me know if there is anything else I should do.

**Comment by oandrew** [comment]
when running speaker-test you should hear some noise in speakers.

could you share the trace file?

**Comment by camper8080** [comment]
Yes, I do hear audio alternating between speakers. It runs for maybe 30 seconds, then stops playing audio, head unit says unsupported. The speaker test continues even though audio is not being played. 
[trace2.log](https://github.com/oandrew/ipod-gadget/files/4155543/trace2.log)

**Comment by camper8080** [comment]
Installed shairport-sync while using hw:1 (card 1: iPodUSB [], device 0: iPod PCM []) in the config and it started playing music for about 15 seconds, the stopped stating unsupported - same issue as above. I was happy to hear music coming out! :)

**Comment by oandrew** [comment]
Just pushes some changes to master - might fix it for you

**Comment by camper8080** [comment]
Playback is uninterrupted with the new changes. Only issue I had was changing sources (from USB (iPod), to radio and back, experienced drops and static. Could still hear music but not well. Not sure if this is something you can fix or not. I can also look at my shairport-sync settings. Attaching another trace if that helps. Ran it for a good 5-10 minutes, switch songs, and sources. Might all be in the trace.

[trace3.log](https://github.com/oandrew/ipod-gadget/files/4157353/trace3.log)

Thanks again for helping me and your time. 👍

**Comment by oandrew** [comment]
i'm glad it worked!

**Comment by camper8080** [comment]
From my last trace, can you see if steering wheel commands are sent? If so, is this something handled by your code or somewhere else? Any tips are appreciated. Thank you.

**Comment by camper8080** [comment]
I see external commands in the log when hitting the forward/back on the steering wheel. Not sure how to get it to change tracks using shairport-sync. The original issue was resolved.

**Comment by camper8080** [comment]
I started having issues with playback. Starts playing audio for about 30 seconds, then stops. Says unsupported. Running iOS 13.5.

[tracelog.txt](https://github.com/oandrew/ipod-gadget/files/4741397/tracelog.txt)

**Comment by Kytech** [comment]
@camper8080 I had been having issues with this in my own car, a 2009 Civic. It was not working until just recently. I pulled down and rebuilt the ipod application from the newest codebase on master and it worked like a charm for audio playback.

From what I've seen in the codebase as I've worked with it, it looks like the code doesn't currently implement track navigation. As of right now, it acknowledges the signal and recognizes it, but it doesn't actually trigger anything on the device that is running the ipod app since there's no code in place to handle the event and forward it to the OS/other apps. This is probably because different apps can be controlled in different ways. Some may respond to a keyboard event, so you'd probably have to write a function to emulate a key press, while others may require communicating with their own API. I have never used shairport-sync, so I can't really provide any guidance relative to that.
